Src/BootCamp.Chapter/Credentials.cs

```diff
@@ -16,6 +25,30 @@ public Credentials(string username, string password)
         public static bool TryParse(string input, out Credentials credentials)
         {
             credentials = default;
+            string[] parts = input.Split(",", 2);
+            if (parts.Length != 2)
+            {
+                return false;
+            }
+
+            string username = parts[0];
+            string password = parts[1];
+
+            credentials = new Credentials(username, password);
+            return true;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
```

Comment:
This check if redundant. If object is null, then obj is Credentials will also be false.